astropy/stats/jackknife.py

```diff
@@ -0,0 +1,197 @@
+    >>> data = np.array([1,2,3,4,5,6,7,8,9,0])
+    >>> resamples = jackknife_resampling(data)
+    >>> resamples
+    array([[ 2.,  3.,  4.,  5.,  6.,  7.,  8.,  9.,  0.],
+           [ 1.,  3.,  4.,  5.,  6.,  7.,  8.,  9.,  0.],
+           [ 1.,  2.,  4.,  5.,  6.,  7.,  8.,  9.,  0.],
+           [ 1.,  2.,  3.,  5.,  6.,  7.,  8.,  9.,  0.],
+           [ 1.,  2.,  3.,  4.,  6.,  7.,  8.,  9.,  0.],
+           [ 1.,  2.,  3.,  4.,  5.,  7.,  8.,  9.,  0.],
+           [ 1.,  2.,  3.,  4.,  5.,  6.,  8.,  9.,  0.],
+           [ 1.,  2.,  3.,  4.,  5.,  6.,  7.,  9.,  0.],
+           [ 1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.,  0.],
+           [ 1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.,  9.]])
+    >>> resamples.shape
+    (10, 9)
+
+    References
+    ----------
+    .. [1] McIntosh, Avery. "The Jackknife Estimation Method".
+        <http://people.bu.edu/aimcinto/jackknife.pdf>
+
+    .. [2] Efron, Bradley. "The Jackknife, the Bootstrap, and other
+        Resampling Plans". Technical Report No. 63, Division of Biostatistics,
+        Stanford University, December, 1980.
+
+    .. [3] Cowles, Kate. "Computing in Statistics: The Jackknife, Lecture 11".
+        <http://homepage.stat.uiowa.edu/~kcowles/s166_2009/lect11.pdf>.
+        September, 2009.
+    """
+
+    n = data.shape[0]
+    assert n > 0, "data must contain at least one measurement"
+
+    resamples = np.empty([n, n-1])
+
+    for i in range(n):
+        resamples[i] = np.delete(data, i)
+
+    return resamples
+
+
+def jackknife_stats(data, statistic, conf_lvl=0.95):
+    """ Performs jackknife estimation on the basis of jackknife resamples.
+
+    This function requires `SciPy <http://www.scipy.org>`_ to be installed.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Original sample (1-D array).
+    statistic : function
+        Any function (or vector of functions) on the basis of the measured
+        data, e.g, sample mean, sample variance, etc. The jackknife estimate of
+        this statistic will be returned.
+    conf_lvl : float
+        Confidence level for the confidence interval of the Jackknife estimate.
+        Must be a real-valued number in (0,1).
+
+    Returns
+    -------
+    estimate : numpy.float64 or numpy.ndarray
+        The i-th element is the bias-corrected "jackknifed" estimate.
+
+    bias : numpy.float64 or numpy.ndarray
+        The i-th element is the jackknife bias.
+
+    std_err : numpy.float64 or numpy.ndarray
+        The i-th element is the jackknife standard error.
+
+    conf_interval : numpy.ndarray
+        If ``statistic`` is single-valued, the first and second elements are
+        the lower and upper bounds, respectively. If ``statistic`` is
+        vector-valued, each column corresponds to the confidence interval for
+        each component of ``statistic``. The first and second rows contain the
+        lower and upper bounds, respectively.
+
+    Examples
+    --------
+    1. Obtain Jackknife resamples:
+
```

Comment:
As the resampling has been moved to inside jackknife_stats, this first example is now no longer needed and can be removed.